File: CS/Classes/DragCellsValuesHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.Utils.Paint;
using DevExpress.Utils;

namespace WindowsApplication1
{
    public class DragCellsValuesHelper
    {

        private SelectedCellsBorderHelper _Helper;

        public GridView View
        {
            get { return _Helper.GridView; }
        }

        private GridCell _SourceGridCell;
        public GridCell SourceGridCell
        {
            get { return _SourceGridCell; }
            set { _SourceGridCell = value; }
        }
        public DragCellsValuesHelper(SelectedCellsBorderHelper selectedCellsHelper)
        {
            _Helper = selectedCellsHelper;
            InitViewEvents();
        }

        private void InitViewEvents()
        {
            View.MouseDown += new MouseEventHandler(View_MouseDown);
            View.MouseUp += new MouseEventHandler(View_MouseUp);
            View.ShowingEditor += new CancelEventHandler(View_ShowingEditor);
            View.SelectionChanged += new DevExpress.Data.SelectionChangedEventHandler(View_SelectionChanged);
        }

        void View_SelectionChanged(object sender, DevExpress.Data.SelectionChangedEventArgs e)
        {
            if (_Helper.IsCopyMode)
                UpdateHint();
        }

        private void UpdateHint()
        {
            ToolTipController.DefaultController.HideHint();
            string text = View.GetFocusedDisplayText();
            if (string.IsNullOrEmpty(text))
                return;
            ToolTipController.DefaultController.ShowHint(text);
        }
        void View_ShowingEditor(object sender, CancelEventArgs e)
        {
            e.Cancel = _Helper.IsCopyMode;
        }

        void View_MouseUp(object sender, MouseEventArgs e)
        {
            if (_Helper.IsCopyMode)
                OnCopyFinished();
            _Helper.IsCopyMode = false;
        }

        private void OnCopyFinished()
        {
            ToolTipController.DefaultController.HideHint();
            CopyCellsValues();
        }


        private void CopyCellsValues()
        {
            object value = View.GetRowCellValue(SourceGridCell.RowHandle, SourceGridCell.Column);
            GridCell[] selectedCells = View.GetSelectedCells();
            foreach (GridCell cell in selectedCells)
            {
                View.SetRowCellValue(cell.RowHandle, cell.Column, value);
            }
        }

        void View_MouseDown(object sender, MouseEventArgs e)
        {
            _Helper.IsCopyMode = _Helper.GetDragRect().Contains(e.Location);
            if (_Helper.IsCopyMode)
                OnStartCopy();
        }

        private void OnStartCopy()
        {
            SourceGridCell = new GridCell(View.FocusedRowHandle, View.FocusedColumn);
        }
    }
}

File: VB/Classes/DragCellsValuesHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.Utils.Paint
Imports DevExpress.Utils

Namespace WindowsApplication1
	Public Class DragCellsValuesHelper

		Private _Helper As SelectedCellsBorderHelper

		Public ReadOnly Property View() As GridView
			Get
				Return _Helper.GridView
			End Get
		End Property

		Private _SourceGridCell As GridCell
		Public Property SourceGridCell() As GridCell
			Get
				Return _SourceGridCell
			End Get
			Set(ByVal value As GridCell)
				_SourceGridCell = value
			End Set
		End Property
		Public Sub New(ByVal selectedCellsHelper As SelectedCellsBorderHelper)
			_Helper = selectedCellsHelper
			InitViewEvents()
		End Sub

		Private Sub InitViewEvents()
			AddHandler View.MouseDown, AddressOf View_MouseDown
			AddHandler View.MouseUp, AddressOf View_MouseUp
			AddHandler View.ShowingEditor, AddressOf View_ShowingEditor
			AddHandler View.SelectionChanged, AddressOf View_SelectionChanged
		End Sub

		Private Sub View_SelectionChanged(ByVal sender As Object, ByVal e As DevExpress.Data.SelectionChangedEventArgs)
			If _Helper.IsCopyMode Then
				UpdateHint()
			End If
		End Sub

		Private Sub UpdateHint()
			ToolTipController.DefaultController.HideHint()
			Dim text As String = View.GetFocusedDisplayText()
			If String.IsNullOrEmpty(text) Then
				Return
			End If
			ToolTipController.DefaultController.ShowHint(text)
		End Sub
		Private Sub View_ShowingEditor(ByVal sender As Object, ByVal e As CancelEventArgs)
			e.Cancel = _Helper.IsCopyMode
		End Sub

		Private Sub View_MouseUp(ByVal sender As Object, ByVal e As MouseEventArgs)
			If _Helper.IsCopyMode Then
				OnCopyFinished()
			End If
			_Helper.IsCopyMode = False
		End Sub

		Private Sub OnCopyFinished()
			ToolTipController.DefaultController.HideHint()
			CopyCellsValues()
		End Sub


		Private Sub CopyCellsValues()
			Dim value As Object = View.GetRowCellValue(SourceGridCell.RowHandle, SourceGridCell.Column)
			Dim selectedCells() As GridCell = View.GetSelectedCells()
			For Each cell As GridCell In selectedCells
				View.SetRowCellValue(cell.RowHandle, cell.Column, value)
			Next cell
		End Sub

		Private Sub View_MouseDown(ByVal sender As Object, ByVal e As MouseEventArgs)
			_Helper.IsCopyMode = _Helper.GetDragRect().Contains(e.Location)
			If _Helper.IsCopyMode Then
				OnStartCopy()
			End If
		End Sub

		Private Sub OnStartCopy()
			SourceGridCell = New GridCell(View.FocusedRowHandle, View.FocusedColumn)
		End Sub
	End Class
End Namespace
